### PR TITLE
Add back beta features to main

### PR DESCRIPTION
**Pull request by rakshith91**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/search/azure-search-documents/azure/search/documents/_generated/_configuration.py [hunk 1]
```diff
@@ -36,15 +39,17 @@ def __init__(
         **kwargs  # type: Any
     ):
         # type: (...) -> None
+        super(SearchClientConfiguration, self).__init__(**kwargs)
+        api_version = kwargs.pop('api_version', "2021-04-30-Preview")  # type: str
```
Comment (xiangyan99) [inline sdk/search/azure-search-documents/azure/search/documents/_generated/_configuration.py:43]: Not DEFAULT_VERSION?

File: sdk/search/azure-search-documents/azure/search/documents/indexes/_search_index_client.py [hunk 2]
```diff
@@ -386,8 +386,8 @@ def delete_synonym_map(self, synonym_map, **kwargs):
         the SynonymMap model must be provided instead of the name. It is enough to provide
         the name of the synonym map to delete unconditionally.
 
-        :param synonym_map: The synonym map name or object to delete
-        :type synonym_map: str or ~azure.search.documents.indexes.models.SynonymMap
+        :param name: The synonym map name or object to delete
```
[the remark this answers is not in the document]
Comment (rakshith91) [reply sdk/search/azure-search-documents/azure/search/documents/indexes/_search_index_client.py:389]: no - bad documentation fix

File: sdk/search/azure-search-documents/azure/search/documents/indexes/_search_indexer_client.py [hunk 3]
```diff
@@ -493,7 +526,7 @@ def get_skillsets(self, **kwargs):
         if kwargs.get('select', None):
             kwargs['select'] = ','.join(kwargs['select'])
         result = self._client.skillsets.list(**kwargs)
-        return result.skillsets
+        return [SearchIndexerSkillset._from_generated(skillset) for skillset in result.skillsets] # pylint:disable=protected-access
```
Comment (xiangyan99) [inline sdk/search/azure-search-documents/azure/search/documents/indexes/_search_indexer_client.py:529]: Is this breaking?

File: sdk/search/azure-search-documents/azure/search/documents/_generated/_configuration.py (hunk 1 above)
Comment (rakshith91) [reply sdk/search/azure-search-documents/azure/search/documents/_generated/_configuration.py:43]: generated code :)

File: sdk/search/azure-search-documents/azure/search/documents/indexes/_search_indexer_client.py [hunk 4]
```diff
@@ -587,8 +621,11 @@ def create_skillset(self, skillset, **kwargs):
 
         """
         kwargs["headers"] = self._merge_client_headers(kwargs.get("headers"))
+        _validate_skillset(skillset)
+        skillset = skillset._to_generated() if hasattr(skillset, '_to_generated') else skillset # pylint:disable=protected-access
 
-        return self._client.skillsets.create(skillset, **kwargs)
+        result = self._client.skillsets.create(skillset, **kwargs)
+        return SearchIndexerSkillset._from_generated(result) # pylint:disable=protected-access
```
Comment (xiangyan99) [inline sdk/search/azure-search-documents/azure/search/documents/indexes/_search_indexer_client.py:628]: Same, breaking?

File: sdk/search/azure-search-documents/azure/search/documents/indexes/_search_indexer_client.py (hunk 3 above)
Comment (rakshith91) [reply sdk/search/azure-search-documents/azure/search/documents/indexes/_search_indexer_client.py:529]: Nope - the namespace for `SearchIndexerSkillset` is not in generated anymore

File: sdk/search/azure-search-documents/azure/search/documents/indexes/_search_indexer_client.py (hunk 4 above)
Comment (rakshith91) [reply sdk/search/azure-search-documents/azure/search/documents/indexes/_search_indexer_client.py:628]: same as above

**Review by xiangyan99 — APPROVED** [review]

**Pull request merged**